shared/crypto/bls_test.go

```diff
@@ -0,0 +1,252 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"testing"
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/pairing/bn256"
+	"go.dedis.ch/kyber/v3/util/random"
+)
+
+func TestBLS(t *testing.T) {
+	msg := []byte("Hello Boneh-Lynn-Shacham")
+	suite := bn256.NewSuite()
+	private, public := NewKeyPair(suite, random.New())
+	sig, err := Sign(suite, private, msg)
+	require.Nil(t, err)
+	err = Verify(suite, public, msg, sig)
+	require.Nil(t, err)
+}
+
+func TestBLSFailSig(t *testing.T) {
+	msg := []byte("Hello Boneh-Lynn-Shacham")
+	suite := bn256.NewSuite()
+	private, public := NewKeyPair(suite, random.New())
+	sig, err := Sign(suite, private, msg)
+	require.Nil(t, err)
+	sig[0] ^= 0x01
+	if Verify(suite, public, msg, sig) == nil {
+		t.Fatal("bls: verification succeeded unexpectedly")
+	}
+}
+
+func TestBLSFailKey(t *testing.T) {
+	msg := []byte("Hello Boneh-Lynn-Shacham")
+	suite := bn256.NewSuite()
+	private, _ := NewKeyPair(suite, random.New())
+	sig, err := Sign(suite, private, msg)
+	require.Nil(t, err)
+	_, public := NewKeyPair(suite, random.New())
+	if Verify(suite, public, msg, sig) == nil {
+		t.Fatal("bls: verification succeeded unexpectedly")
+	}
+}
+
+func TestBLSAggregateSignatures(t *testing.T) {
+	msg := []byte("Hello Boneh-Lynn-Shacham")
+	suite := bn256.NewSuite()
+	private1, public1 := NewKeyPair(suite, random.New())
+	private2, public2 := NewKeyPair(suite, random.New())
+	sig1, err := Sign(suite, private1, msg)
+	require.Nil(t, err)
+	sig2, err := Sign(suite, private2, msg)
+	require.Nil(t, err)
+	aggregatedSig, err := AggregateSignatures(suite, sig1, sig2)
+	require.Nil(t, err)
```

Comment:
Can you update the test to also verify that the `sizeof(aggregatedSignature)=sizeof(sig1)` and `sizeof(aggregatedSignature)=sizeof(sig2)`